docs/flows/flows.puml

```diff
@@ -30,17 +30,16 @@ database Snapshot as snapshot
 end box
 
 box "(onchain)" #Lavender
+collections Proxy as proxy
```

Comment:
question: Is there only one proxy contract?

As there are multiple contract (Bond, Rewards, Insurance, Treasuries), do they all have individual proxies, or is this a shared mechanism?

Reply:
Each has its own proxy instance deployed, but Proxy and ProxyAdmin contracts are code equal. I don't indicate separate proxy entities because its same for all, deploying same proxy contract and proxy admin to not overload with additional special proxy entities.

Reply:
>Proxy and ProxyAdmin contracts are code equal. I don't indicate separate proxy entities because its same for all,

Thought that might be the case. Unfortunately that is not the reading of the UML. The fact that the proxy and proxyadmin are a pattern and not single actors (as each of the other actors are) needs to be conveyed. 

I'd be inclined to move everything related to the proxy into a separate PUML file, and remove the `proxy` and `proxyadmin` from this file ...otherwise creative use of notes could be used 🤔 
@shahankhatch any input on this matter?

Reply:
Moved everything already, though I'd still need to indicate that we deploy `proxy` and `proxyadmin`, maybe without entities or just use `bitdaotreasury`, `bitdaogovernance` and treat them as proxy as it somewhat would be.

Reply:
The suggested approach by @shahankhatch about replacing the explicit proxy interactions with notes is a good one.

Reply:
But we still want to make proxy deployment process explicit as they are separate steps. The last part where we make upgrade interaction is a note indeed.